editor/src/core/shared/import/proejct-health-check/check-utopia-requirements.ts

```diff
@@ -0,0 +1,128 @@
+} from '../../../../components/assets'
+import type { ProjectContentTreeRoot } from 'utopia-shared/src/types'
+import { codeFile, isTextFile, RevisionsState } from '../../project-file-types'
+import { notifyCheckingRequirement, notifyResolveRequirement } from './utopia-requirements-service'
+import { RequirementResolutionResult } from './utopia-requirements-types'
+import { applyToAllUIJSFiles } from '../../../../core/model/project-file-utils'
+
+export function checkAndFixUtopiaRequirements(
+  parsedProjectContents: ProjectContentTreeRoot,
+): ProjectContentTreeRoot {
+  let projectContents = parsedProjectContents
+  // check package.json
+  projectContents = checkAndFixPackageJson(projectContents)
+  // check language
+  projectContents = checkProjectLanguage(projectContents)
+  // check react version
+  checkReactVersion(projectContents)
+  return projectContents
+}
+
+function getPackageJson(
+  projectContents: ProjectContentTreeRoot,
+): { utopia?: Record<string, string>; dependencies?: Record<string, string> } | null {
+  const packageJson = packageJsonFileFromProjectContents(projectContents)
+  if (packageJson != null && isTextFile(packageJson)) {
+    return JSON.parse(packageJson.fileContents.code)
+  }
+  return null
+}
+
+function checkAndFixPackageJson(projectContents: ProjectContentTreeRoot): ProjectContentTreeRoot {
+  notifyCheckingRequirement('packageJsonEntries', 'Checking package.json')
+  const parsedPackageJson = getPackageJson(projectContents)
+  if (parsedPackageJson == null) {
+    notifyResolveRequirement(
+      'packageJsonEntries',
+      RequirementResolutionResult.Critical,
+      'The file package.json was not found',
+    )
+    return projectContents
+  }
+  if (parsedPackageJson.utopia == null) {
+    parsedPackageJson.utopia = {
+      'main-ui': 'utopia/storyboard.js',
+    }
+    const result = addFileToProjectContents(
+      projectContents,
+      '/package.json',
+      codeFile(
+        JSON.stringify(parsedPackageJson, null, 2),
+        null,
+        0,
+        RevisionsState.CodeAheadButPleaseTellVSCodeAboutIt,
+      ),
+    )
+    notifyResolveRequirement(
+      'packageJsonEntries',
+      RequirementResolutionResult.Fixed,
+      'Fixed utopia entry in package.json',
+    )
+    return result
+  } else {
+    notifyResolveRequirement(
+      'packageJsonEntries',
+      RequirementResolutionResult.Found,
+      'Valid package.json found',
+    )
+  }
+
+  return projectContents
+}
+
+function checkProjectLanguage(projectContents: ProjectContentTreeRoot): ProjectContentTreeRoot {
+  notifyCheckingRequirement('language', 'Checking project language')
+  let jsCount = 0
+  let tsCount = 0
+  applyToAllUIJSFiles(projectContents, (filename, uiJSFile) => {
+    if (filename.endsWith('.ts') || filename.endsWith('.tsx')) {
+      tsCount++
+    } else if (filename.endsWith('.js') || filename.endsWith('.jsx')) {
+      jsCount++
+    }
+    return uiJSFile
+  })
+  if (tsCount > jsCount) {
```

Comment:
Would we ever want to pass this check if `tsCount > 0`? I can't think of a reason why we would 🤔

Reply:
yes I was trying to be a bit permissive here with an rough heuristic, in case they have several `.ts` files that are maybe unrelated to the main app flow. But if we're giving them the option to "continue anyway" even with errors, then they can maybe mark this as a non-issue if that's the case. I'll change to `>0`.